kaos/parser.py

```diff
@@ -0,0 +1,130 @@
+""""Handles the reading and parsing of ephemeris files.
+Parsed files are stored in the DB.
+"""
+import os
+from collections import namedtuple
+from kaos.models import DB, OrbitRecords, SatelliteInfo, OrbitSegments
+
+OrbitPoint = namedtuple('OrbitPoint', 'time, pos, vel')
+
+def add_segment_to_db(orbit_data, satellite_id):
+    """Add the given segment to the database.
+    We create a new entry in the Segment DB that holds
+    - segment_id
+    - segment_start
+    - segment_end
+    - satellite_id
+
+    This segment element is also used to index a group of
+    rows in the Orbits DB. This lets us know that the orbit
+    data belongs to a given segment. This is because
+    we cannot perform interpolation using points in
+    different segments."""
+
+    # TODO Validate satellite_id
+
+    segment_start = orbit_data[0].time
+    segment_end = orbit_data[-1].time
+
+    """create segment entry.
+    Retrieve segment ID and insert
+    it along with data into Orbit db"""
+    segment = OrbitSegments(platform_id=satellite_id, start_time=segment_start,
+                            end_time=segment_end)
+
+    # TODO Check if a segment exists that overlaps this segment
+    segment.save()
+    DB.session.commit()
+
+    for orbit_point in orbit_data:
+        # TODO Validate uniqueness for this platform
+        orbit_record = OrbitRecords(platform_id=satellite_id, segment_id=segment.segment_id,
+                                    time=orbit_point.time, position=orbit_point.pos,
+                                    velocity=orbit_point.vel)
+        orbit_record.save()
+
+    DB.session.commit()
+
+def parse_ephemeris_file(file_handle):
+    """Parse the given ephemeris file and store
+    the orbital data into the DB"""
+
+    #TODO: pre-process the file. Verify file formatting is as we expect
+
+    sat = SatelliteInfo(platform_name=os.path.splitext(file_handle)[0])
+    sat.save()
+    DB.session.commit()
+
+    with open(file_handle, "rU") as f:
+        """A list containing each of the 15 or so
```

Comment:
Comments as to what the variable is should not be there. The name should make it obvious ( which it does )